File: 프로그래머스/레벨3/12979. 기지국 설치/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int cover = 2 * w + 1;
        int result = 0;
        
        // 좌측 끝 처리
        int left = stations[0] - 1 - w;
        if (left > 0) result += Math.ceil((double)left / cover);
        
        // 중간 구간 처리
        int start = stations[0];
        for (int i = 1; i < stations.length; i++) {
            int end = stations[i];
            int mid = end - start - cover;
            if (mid > 0) result += Math.ceil((double)mid / cover);
            
            start = end;
        }
        
        // 우측 끝 처리
        int right = n - stations[stations.length - 1] - w;
        if (right > 0) result += Math.ceil((double)right / cover);
        
        return result;
    }
}

File: 프로그래머스/레벨3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
    const cover = 2 * w + 1;
    let result = 0;
    
    // 좌측 끝 처리
    const left = stations[0] - w -1
    if (left > 0) result += Math.ceil(left / cover);
    
    // 중간 구간 처리
    let start = stations[0];
    for (let i = 1; i < stations.length; i++) {
        let end = stations[i];
        let mid = end - start - cover;
        if (mid > 0) result += Math.ceil(mid / cover);
        
        start = end;
    }
    
    // 우측 끝 처리
    const right = n - stations[stations.length - 1] - w;
    if (right > 0) result += Math.ceil(right / cover);
    
    return result;
}